.editorconfig

```diff
@@ -1,12 +1,17 @@
 root = true
 
-[*.{py,c,cpp,h,rst,md,yml}]
+[*.{py,c,cpp,h,rst,md,yml,json}]
 trim_trailing_whitespace = true
 insert_final_newline = true
 indent_style = space
 
-[*.{py,c,cpp,h}]
+[*.{py,c,cpp,h,json}]
 indent_size = 4
```

Comment:
Formatting JSON files does not seem necessary for this repo.  I have only found a few, and they seem to be imported from external sources or maintained without issue.

Reply:
We have 11 json files in the repo: https://github.com/python/cpython/search?q=language%3Ajson

I don't think `json` support is any different from `yml`, which we already have.

Reply:
I know, that’s what I called a few, and you’re ignoring my second note that they are mostly copied from external sources.

I am -1 on this change.  It odes not solve any problem.

Reply:
Sorry, not meant to ignore your words at all! 🙂 
It was more like a personal research, because I know that are not-so-many `json` files, but I was not sure about the exact amount.

I also agree that this is not a real "problem" at all, no one was ever hurt by the wrong number of spaces in a `json` file (at least I really hope so!).

But, the initial reason I've created this PR was this file in its current form: https://github.com/python/cpython/blob/main/.github/problem-matchers/msvc.json

I was reading through it and asked myself: why does it have mixed formatting? What is the rule? I looked at `.editorconfig`, but didn't find any. And since I was also fixing `.editorconfig` files in other `python/*` repositories, I've decided to create a PR that can potentially solve this *tiny* problem in the future.

But, I don't have strong feelings about a so-little-change, so I / you / anyone-else can totally close this PR if needed 👍 

Thanks a lot for your feedback and review, TIL about `tabstop`!

Reply:
With an old, stable codebase like CPython, the team prefers to solve real problems (reported by end users or core developers) and avoid cosmetic changes that don’t provide a concrete improvement, may introduce errors and muddle history (git log is not very good, not to mention github history that doesn’t even follow renames).  Consistency in itself and for itself is not a goal!  Thanks for your efforts, they will be very appreciated for other kinds of PRs!